### DISCUSSION TITLE
Add examples setting type of initially empty state

### DISCUSSION THREAD

**Discussion by theoephraim** [vuejs/pinia#482] [Ideas]
Just started using pinia and loving it so far.

One quick suggestion - add examples showing how to set TypeScript types on state properties that may initially be empty or are not simple enough to infer.

For example
```
export const useMyStore = defineStore({
  state: () => ({
    myInitiallyNullItem: null as SomeType | null,
    myFancyObject: {} as SomeObjectDescription,
...
```

I assume that is the suggested way to do this? I stumbled upon it in some random issue and it seems obvious now, but I think would definitely be helpful to mention it in the docs.

**Comment by posva** [comment]
Yes, that's the recommended way. Definitely worth adding to the documentation in the state page! If you want you can do a PR

**Reply by theoephraim** [reply]
Great. Thanks for confirming. Will add a PR!
